packages/indexer-cli/src/commands/indexer/rules/set.ts

```diff
@@ -61,23 +63,31 @@ module.exports = {
         return
       }
 
-      // Turn the array into an object, add `identifier` and `identifierType` keys
-      const inputRule = parseIndexingRule({
-        ...Object.fromEntries([...partition(2, 2, kvs)]),
-        identifier,
-        identifierType,
-      })
-
-      if (inputRule.parallelAllocations && inputRule.parallelAllocations >= 2) {
-        print.error(
-          'Parallel allocations are soon to be fully deprecated. Please set parallel allocations to 1 for all your indexing rules',
+      // Turn input into an indexing rule object, add `identifier` and `identifierType` keys
+      try {
+        const inputRule = parseIndexingRule({
+          ...Object.fromEntries([...partition(2, 2, kvs)]),
+          identifier,
+          identifierType,
+        })
+        if (inputRule.parallelAllocations && inputRule.parallelAllocations >= 2) {
+          print.error(
+            'Parallel allocations are soon to be fully deprecated. Please set parallel allocations to 1 for all your indexing rules',
+          )
+          process.exitCode = 1
+        }
+        const rule = await setIndexingRule(client, inputRule)
+        printIndexingRules(print, outputFormat, identifier, rule, [])
+      } catch (error) {
+        // Failed to parse input, make suggestions
+        const valid_commands = Object.keys(
+          (await indexingRules(client,false))[0],
```

Comment:
or I could call `await indexingRule(client,'global', false)`? although that returns type potentially null that needs more conditionals to handle. Getting index 0 here is basically the global rule which is ensured to exist. Not sure if we need to worry about not having an empty list